### ISSUE TITLE
Updated notifier breaks `maizzle build` in monorepos.

### ISSUE THREAD

**Issue by joshmanders** [maizzle/cli#198]
I've had no issues before but haven't updated in a while, being as we're still on `v4` I just wholesale updated from `4.0.2` to `4.4.4` and now `maizzle build --bin ....` breaks because `updateNotifier` doesn't take the `--bin` flag into consideration, it still looks for `./node_modules` which doesn't exist in many monorepos.

Looks like this issue was brought in here: https://github.com/maizzle/cli/commit/847d7f3abb1610b3552e676e65d974deae1c0250

**Comment by joshmanders** [comment]
@cossssmin any input on this? I can submit a PR adding the try catch back... Also you could use `require.resolve('@maizzle/framework/package.json')` and that will use the internal module resolver to find the package the same way `require('@maizzle/framework')` would. This finding it in monorepos.

**Comment by cossssmin** [comment]
Hey Josh, sorry for the delay, busy times... Yes, please open a PR and we take it from there, thank you!

**Comment by joshmanders** [comment]
@cossssmin 
> Hey Josh, sorry for the delay, busy times... Yes, please open a PR and we take it from there, thank you!

No worries, big fan of maizzle, thank you for what you've already done. I've opened a PR #200

**Comment by cossssmin** [comment]
Sorry need to reopen this and revert the change, it's breaking Maizzle right now...

**Comment by cossssmin** [comment]
Error stack ref:

```
node:internal/modules/cjs/loader:995
  const err = new Error(message);
              ^

Error: Cannot find module '@maizzle/framework/src'
Require stack:
- F:\DEV\open-source\maizzle\cli\src\index.js
- F:\DEV\open-source\maizzle\cli\bin\maizzle
    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Function.resolve (node:internal/modules/cjs/helpers:109:19)
    at Command.<anonymous> (F:\DEV\open-source\maizzle\cli\src\index.js:47:42)
    at Command.listener [as _actionHandler] (F:\DEV\open-source\maizzle\cli\node_modules\commander\lib\command.js:482:17)
    at F:\DEV\open-source\maizzle\cli\node_modules\commander\lib\command.js:1283:65
    at Command._chainOrCall (F:\DEV\open-source\maizzle\cli\node_modules\commander\lib\command.js:1177:12)
    at Command._parseCommand (F:\DEV\open-source\maizzle\cli\node_modules\commander\lib\command.js:1283:27)
    at F:\DEV\open-source\maizzle\cli\node_modules\commander\lib\command.js:1081:27
    at Command._chainOrCall (F:\DEV\open-source\maizzle\cli\node_modules\commander\lib\command.js:1177:12)
    at Command._dispatchSubcommand (F:\DEV\open-source\maizzle\cli\node_modules\commander\lib\command.js:1077:23) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'F:\\DEV\\open-source\\maizzle\\cli\\src\\index.js',
    'F:\\DEV\\open-source\\maizzle\\cli\\bin\\maizzle'
  ]
}
```

**Comment by cossssmin** [comment]
This is an issue when running the Maizzle CLI globally.

So if you have installed it with `npm install -g @maizzle/cli` and run `maizzle build` in the project root, you'll get an error similar to the one above; Node can't resolve the framework location because it has no way of knowing where it is in a 'global' context.

Running the NPM scripts, like `npm run build`, works fine because Node can resolve `@maizzle/framework` inside your project.

You can pin `@maizzle/cli` to `1.5.3` but for now I have reverted it until we find a solution. 

Most probably we'll need a try/catch and fall back to the old way, but I'm open to suggestions :)

**Comment by joshmanders** [comment]
> This is an issue when running the Maizzle CLI globally.
> 
> So if you have installed it with `npm install -g @maizzle/cli` and run `maizzle build` in the project root, you'll get an error similar to the one above; Node can't resolve the framework location because it has no way of knowing where it is in a 'global' context.
> 
> Running the NPM scripts, like `npm run build`, works fine because Node can resolve `@maizzle/framework` inside your project.
> 
> You can pin `@maizzle/cli` to `1.5.3` but for now I have reverted it until we find a solution.
> 
> Most probably we'll need a try/catch and fall back to the old way, but I'm open to suggestions :)

Oh interesting, I haven't ran a cli from global scope in a long time. I'll see if I can come up with a solution but for now pin to 1.5.3.

**Comment by cossssmin** [comment]
Hi @joshmanders, can you try the [`v1.5.5-01` prerelease](https://github.com/maizzle/cli/releases/tag/v1.5.5-0) and see how that works for you? Tagged it under `next` in NPM as well, so you can also get it with `npm i @maizzle/cli@next`.

Used [`resolve-cwd`](https://github.com/sindresorhus/resolve-cwd) to try and resolve the framework executable from the current directory, should work fine now.

**Comment by joshmanders** [comment]
> Hi @joshmanders, can you try the [`v1.5.5-01` prerelease](https://github.com/maizzle/cli/releases/tag/v1.5.5-0) and see how that works for you? Tagged it under `next` in NPM as well, so you can also get it with `npm i @maizzle/cli@next`.
> 
> Used [`resolve-cwd`](https://github.com/sindresorhus/resolve-cwd) to try and resolve the framework executable from the current directory, should work fine now.

Yes I will try it later this afternoon when I get back to my desk and report back. Thank you!

**Comment by joshmanders** [comment]
@cossssmin works perfectly!

**Comment by cossssmin** [comment]
Sweet!

Will release it next week since I’ll be away for Easter here 👍

**Comment by joshmanders** [comment]
I can stay on the old version that worked until you release so no worries. Have a happy easter!

**Comment by cossssmin** [comment]
Published v1.5.5, give it a try and let me know how it goes 👍

https://github.com/maizzle/cli/releases/tag/v1.5.5
